### ISSUE TITLE
Could not find or use auto-linked framework 'Flutter' in Xcode extension target

### ISSUE THREAD

**Issue by rafaelsetragni** [flutter/flutter#103344]
Hi everyone!

Im trying to use a `Notification Service Extension `on iOS to show rich push notifications. Also i created a local notification plugin called [Awesome Notifications](https://pub.dev/packages/awesome_notifications) and im want to reuse this logic inside the Target extension.

Just for better comprehension, what im trying  to do is exactly what One Signal did for their plugin:
https://documentation.onesignal.com/docs/flutter-sdk-setup#step-3---add-an-ios-service-extension-ios-apps-only

When I created the app for the first time, without the target extension, and add the target extension later, everything compiles and runs ok on the real device. But, when the project is cleaned or I tried to compile with release mode, im starting to get compilation erros about some unknown Flutter symbols, such as:

![image](https://user-images.githubusercontent.com/40064496/167436087-9083818f-eeb8-4f42-a660-4994ab9c11ed.png)

What I realized was the Flutter engine wasn't compiled at the moment when the target extension is being built, because the main app depends on target extension, so the target extension must be built first. But the flutter library is only compiled on main app/target.

This way, these flutter symbols are not getting recognized on the target extension for cases where the project was cleaned and is being recreated from ground up (something that happens on each `pod install` or `flutter run --release`). But, if the app was already built once without the target extension, the compiler becomes able to find the flutter symbols from cached files, and it compiles the whole project successfully.

How do I correctly use my flutter plugin's native classes inside a target extension? There is a way to create a pod file modification to do that automatically for each build?

This is my pod.file:
```Pod
# Uncomment this line to define a global platform for your project
platform :ios, '10.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

target 'AwesomeServiceExtension' do
  use_frameworks!
  use_modular_headers!

  flutter_install_ios_engine_pod File.dirname(File.realpath(__FILE__))
  pod 'awesome_notifications_fcm', :path => '.symlinks/plugins/awesome_notifications_fcm/ios'
  #flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      if target.name.start_with? 'Awesome'
        config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'NO'
      end
    end
    flutter_additional_ios_build_settings(target)
  end
end
```

## Steps to Reproduce

To make it compile successfully:
1. Run the app without target extensions
2. Add the notification service target extension
3. modify pod.file to include the plugin library into the new target extension:
```
target 'AwesomeServiceExtension' do
  use_frameworks!
  use_modular_headers!

  flutter_install_ios_engine_pod File.dirname(File.realpath(__FILE__))
  pod 'awesome_notifications_fcm', :path => '.symlinks/plugins/awesome_notifications_fcm/ios'
  #flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      if target.name.start_with? 'Awesome'
        config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'NO'
      end
    end
    flutter_additional_ios_build_settings(target)
  end
end
```
4. add the plugin import (this case 'awesome_notifications') inside the extension source code file: 
![image](https://user-images.githubusercontent.com/40064496/167435320-17f426e1-0708-499a-a2cc-7d53d0c1c210.png)

5. Rerun the app without clean it.

But, to make the errors appears:
1 . Redo the steps above without running the app previously, excluding the target extension (step 1)

## Logs

```terminal
Xcode's output:
↳
    Writing result bundle at path:
        /var/folders/6d/c48hfkfd7rnfnk8sc9gpc8km0000gn/T/flutter_tools.O0p66k/flutter_ios_build_temp_dirpFtrWY/temporary_xcresult_bundle

    ld: warning: Could not find or use auto-linked framework 'Flutter'
    Undefined symbols for architecture arm64:
      "_OBJC_CLASS_$_FlutterCallbackCache", referenced from:
          objc-class-ref in awesome_notifications(DartBackgroundExecutor.o)
          objc-class-ref in awesome_notifications_fcm(DartFcmBackgroundExecutor.o)
      "_OBJC_CLASS_$_FlutterEngine", referenced from:
          objc-class-ref in awesome_notifications(DartBackgroundExecutor.o)
          objc-class-ref in awesome_notifications_fcm(DartFcmBackgroundExecutor.o)
      "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
          objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
          objc-class-ref in awesome_notifications(DartBackgroundExecutor.o)
          objc-class-ref in awesome_notifications_fcm(SwiftAwesomeNotificationsFcmPlugin.o)
          objc-class-ref in awesome_notifications_fcm(DartFcmBackgroundExecutor.o)
      "_OBJC_CLASS_$_FlutterStandardTypedData", referenced from:
          objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
      "_OBJC_CLASS_$_FlutterError", referenced from:
          objc-class-ref in awesome_notifications(SwiftAwesomeNotificationsPlugin.o)
          objc-class-ref in awesome_notifications(DartBackgroundExecutor.o)
          objc-class-ref in awesome_notifications_fcm(SwiftAwesomeNotificationsFcmPlugin.o)
          objc-class-ref in awesome_notifications_fcm(DartFcmBackgroundExecutor.o)
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in dependency order
    /Users/carda/Documents/GitHub/Plugins/awesome_notifications_fcm/example/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 15.4.99.
    (in target 'Toast' from project 'Pods')

    Result bundle written to path:
        /var/folders/6d/c48hfkfd7rnfnk8sc9gpc8km0000gn/T/flutter_tools.O0p66k/flutter_ios_build_temp_dirpFtrWY/temporary_xcresult_bundle


Could not build the precompiled application for the device.
Error (Xcode): Undefined symbol: _OBJC_CLASS_$_FlutterCallbackCache


Error (Xcode): Undefined symbol: _OBJC_CLASS_$_FlutterEngine


Error (Xcode): Undefined symbol: _OBJC_CLASS_$_FlutterMethodChannel


Error (Xcode): Undefined symbol: _OBJC_CLASS_$_FlutterStandardTypedData


Error (Xcode): Undefined symbol: _OBJC_CLASS_$_FlutterError



Error running application on iPhone de Rafael.
```

My flutter doctor:
```
[✓] Flutter (Channel stable, 2.10.5, on macOS 12.3.1 21E258 darwin-arm, locale en-BR)
    • Flutter version 2.10.5 at /Users/carda/Desenvolvimento/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5464c5bac7 (3 weeks ago), 2022-04-18 09:55:37 -0700
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/carda/Library/Android/sdk
    • Platform android-32, build-tools 32.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • iPhone de Rafael (mobile) • 00008020-000C05010C53802E • ios            • iOS 15.4.1 19E258
    • Chrome (web)              • chrome                    • web-javascript • Google Chrome 101.0.4951.54

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. I am going to keep this issue open and label it for further insights from the team on your question below:

`How do I correctly use my flutter plugin's native classes inside a target extension? There is a way to create a pod file modification to do that automatically for each build?`

/cc @jmagman

**Comment by jmagman** [comment]
Can you `flutter clean` and then run again with the `--verbose` flag?

Off the top of my head I bet you need to set the base configuration for your extension target to match the Flutter app target to get the build settings in the Generated.xcconfig that Flutter generates.

![Screen Shot 2022-05-10 at 12 31 26 PM](https://user-images.githubusercontent.com/682784/167707639-1524cf38-fe8f-4084-b704-58ae97863b0c.png)

**Comment by rafaelsetragni** [comment]
I did it, clean everything with `flutter clean` command, then I changed the pod configuration as described in the last post (same as the main target) and finally executed `flutter run --verbose`. But unfortunately I got the same results.

As you requested, there is my output from `flutter run --verbose` command, just after execute `flutter clean`:
[awesome-debug-run-verbose.txt](https://github.com/flutter/flutter/files/8665609/awesome-debug-run-verbose.txt)

Remembering that the target extension must use a flutter method inside. This is necessary because in my Awesome Notifications plugin I made an option to load notification images based on flutter assets. I did it because it is the most common way to use images in flutter apps, so I thought it would make more sense to enable this feature for flutter developers. Because of that, I need to use the flutter library inside the target extension.

**Comment by jmagman** [comment]
> Off the top of my head I bet you need to set the base configuration for your extension target to match the Flutter app target to get the build settings in the Generated.xcconfig that Flutter generates (in addition to the `flutter_install_ios_engine_pod` in the Podfile target, which you figured out).
> 
> ![Screen Shot 2022-05-10 at 12 31 26 PM](https://user-images.githubusercontent.com/682784/167707639-1524cf38-fe8f-4084-b704-58ae97863b0c.png)

Did you try this?

**Comment by jmagman** [comment]
(Probably a duplicate of https://github.com/flutter/flutter/issues/71262)

**Comment by rafaelsetragni** [comment]
> > Off the top of my head I bet you need to set the base configuration for your extension target to match the Flutter app target to get the build settings in the Generated.xcconfig that Flutter generates (in addition to the `flutter_install_ios_engine_pod` in the Podfile target, which you figured out).
> > ![Screen Shot 2022-05-10 at 12 31 26 PM](https://user-images.githubusercontent.com/682784/167707639-1524cf38-fe8f-4084-b704-58ae97863b0c.png)
> 
> Did you try this?

Sure! I did the same as you:

![image](https://user-images.githubusercontent.com/40064496/168143568-58c69595-35b6-40e4-8716-c69308e783bc.png)

Also, I created a simple example app with the problem reproduced:
https://github.com/rafaelsetragni/Issue_target_extension

> (Probably a duplicate of #71262)

Honestly, I didn't understand the problem cause to know if it's the same case, but I'll keep trying to solve it with her solution steps. If I find a solution I will let you know.

**Comment by jmagman** [comment]
Thanks for the reproduction app, that was very helpful.  With a normal Flutter app:
1. The plugins are built, linking against the framework in your Flutter SDK. If you look in `ios/Pods/Pods.xcodeproj/project.pbxproj` this is done here:
```
				"FRAMEWORK_SEARCH_PATHS[sdk=iphoneos*]" = (
					"\"/Users/magder/Projects/flutter/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64\"",
					"$(inherited)",
				);
				"FRAMEWORK_SEARCH_PATHS[sdk=iphonesimulator*]" = (
					"\"/Users/magder/Projects/flutter/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_x86_64-simulator\"",
					"$(inherited)",
				);
```
and it's hooked up by `flutter_additional_ios_build_settings` for all pods.
2. The app runs the script build phase `Run Script` which calls `/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build`.  This copies the Flutter framework out of the SDK and into Xcode's build products directory.
3. The app links on the framework in the build products directory.

Your extension is building between 1 and 2, after plugins, but before the framework is copied in.  So you can solve this one of two ways (pick one, don't need both):

1. Add the `Run Script` build phase to your extension to run before "Compile Sources".  This will copy the Flutter framework to the built products directory so your extension can find it.
<img src="https://user-images.githubusercontent.com/682784/168178303-7d878f49-a90e-4855-a5d5-77fecdeb8afd.png" width=800>


2. OR add the framework search paths to your extension to find the framework in the SDK, so manually do what `flutter_additional_ios_build_settings` does.
![Screen Shot 2022-05-12 at 3 39 18 PM](https://user-images.githubusercontent.com/682784/168179189-acf6baa3-519b-4f6e-a6b5-396435a3d1d8.png)
![Screen Shot 2022-05-12 at 3 39 25 PM](https://user-images.githubusercontent.com/682784/168179186-c57166e8-dd59-413b-9e85-91ce839b289e.png)

**Framework Search Paths > Debug > Any iOS Simulator SDK** add `"${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_x86_64-simulator"`

**Debug > Any iOS SDK** add `"${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64"`

**Profile and Release > Any iOS Simulator SDK** add `"${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios-release/Flutter.xcframework/ios-arm64_x86_64-simulator"`

**Profile and Release > Any iOS SDK** add `"${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios-release/Flutter.xcframework/ios-arm64"`



If you do this correctly, in `ios/Runner.xcodeproj/project.pbxproj` you will see in Debug:
```
				"FRAMEWORK_SEARCH_PATHS[sdk=iphoneos*]" = (
					"$(inherited)",
					"\"${PODS_CONFIGURATION_BUILD_DIR}/awesome_notifications\"",
					"\"${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64\"",
				);
				"FRAMEWORK_SEARCH_PATHS[sdk=iphonesimulator*]" = (
					"$(inherited)",
					"\"${PODS_CONFIGURATION_BUILD_DIR}/awesome_notifications\"",
					"\"${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_x86_64-simulator\"",
				);
```
and in Profile/Release:
```
				"FRAMEWORK_SEARCH_PATHS[sdk=iphoneos*]" = (
					"$(inherited)",
					"\"${PODS_CONFIGURATION_BUILD_DIR}/awesome_notifications\"",
					"\"${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios-release/Flutter.xcframework/ios-arm64\"",
				);
				"FRAMEWORK_SEARCH_PATHS[sdk=iphonesimulator*]" = (
					"$(inherited)",
					"\"${PODS_CONFIGURATION_BUILD_DIR}/awesome_notifications\"",
					"\"${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios-release/Flutter.xcframework/ios-arm64_x86_64-simulator\"",
				);
```
I tried each of these in your test app and it built.

**Comment by rafaelsetragni** [comment]
Bingo! It built like a charm using the first solution!
Not only that, I automated the configuration process using PodFIle scripts:

First, I created a PodFile script containing all the methods that I need:
```rb
require 'xcodeproj'

def install_awesome_fcm_ios_pod_target(application_path = nil)
    # defined_in_file is set by CocoaPods and is a Pathname to the Podfile.
    application_path ||= File.dirname(defined_in_file.realpath) if self.respond_to?(:defined_in_file)
    raise 'Could not find application path in install_awesome_fcm_ios_pod_target' unless application_path
    
    flutter_install_ios_engine_pod application_path
    pod 'awesome_notifications', :path => File.join('.symlinks', 'plugins', 'awesome_notifications', 'ios')
end

def update_awesome_fcm_service_target(xcodeproj_path)
     target_name = 'AwesomeServiceExtension'
     build_phase_name = "Awesome Notifications Build Phase"
     flutter_build_script = '/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build'
     desired_phase_index = 1
     
     puts "Updating build phase order of '#{build_phase_name}' in #{xcodeproj_path}, #{target_name} \n"
     project = Xcodeproj::Project.open(File.join(xcodeproj_path, 'Runner.xcodeproj'))
     target = project.targets.select { |t| t.name == target_name }.first
     if target.nil?
         raise "AwesomeServiceExtension is missing. You need to create a notification service extension called 'AwesomeServiceExtension' to properly use awesome_notifications_fcm\n"
     end
     
     cocoapods_index = target.build_phases.index { |b|
         name = b.name if b.respond_to? :name
         name == build_phase_name
     }
     
     if cocoapods_index.nil?
         phase = target.new_shell_script_build_phase(build_phase_name)
         phase.shell_script = flutter_build_script
         
         cocoapods_index = target.build_phases.index { |b|
             name = b.name if b.respond_to? :name
             name == build_phase_name
         }
     end
     
     if cocoapods_index != desired_phase_index
         puts "#{target_name}: '#{build_phase_name}' index was: #{cocoapods_index}, moving to index #{desired_phase_index}.\n"
         target.build_phases.move_from(cocoapods_index, desired_phase_index) # move further to the top
     end
     
     target.build_configurations.each do |config|
         config.build_settings['ENABLE_BITCODE'] = 'NO'
     end
     
     project.save
end
```


...and then I reused it, modifying the application PodFile ending:
```rb
flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end

################  Awesome Notifications FCM pod steps   ###################
awesome_pod_file = File.expand_path(File.join('plugins', 'awesome_notifications_fcm', 'ios', 'Scripts', 'AwesomeFcmPodFile'), '.symlinks')
require awesome_pod_file
target 'AwesomeServiceExtension' do
  use_frameworks!
  use_modular_headers!

  install_awesome_fcm_ios_pod_target File.dirname(File.realpath(__FILE__))
end
update_awesome_fcm_service_target File.dirname(File.realpath(__FILE__))
################  Awesome Notifications FCM pod steps   ###################
```


...but this part I couldn't automated yet:

![image](https://user-images.githubusercontent.com/40064496/168143568-58c69595-35b6-40e4-8716-c69308e783bc.png)

Do you know if there is a way to also include this change in PodFile's automated script, preventing developers from doing this themselves?

**Comment by jmagman** [comment]
> Do you know if there is a way to also include this change in PodFile's automated script, preventing developers from doing this themselves?

What do you mean "developers", are you trying to productize the extension?  You also don't need all that code to handle build phases manually, CocoaPods supports `script_phase ... :execution_position => :before_compile`

**Comment by rafaelsetragni** [comment]
> > Do you know if there is a way to also include this change in PodFile's automated script, preventing developers from doing this themselves?
> 
> 
> 
> What do you mean "developers", are you trying to productize the extension?  You also don't need all that code to handle build phases manually, CocoaPods supports `script_phase ... :execution_position => :before_compile`

Yes, not only the extension, bit whole solution. I've created Awesome Notifications plugin for flutter and im trying to let this plugin as easy as possible to any Flutter developer, in any level. A minimum complexity on configuration phase brings a lot of issues to the developers.

**Comment by jmagman** [comment]
Since you're the maintainer, you should be able to change the `awesome_notifications` podspec to run the command for them by leveraging the generated `flutter_export_environment.sh` environment variables, which are the same as are in the generated xcconfig (that base configuration) part:
```ruby
  s.script_phase = { :name => 'Copy Flutter', :script => 'source "${SRCROOT}/../Flutter/flutter_export_environment.sh" && "${FLUTTER_ROOT}/packages/flutter_tools/bin/xcode_backend.sh" build', :execution_position => :before_compile }
```
Then your users don't need to do anything in their Podfile except `pod awesome_notifications` in the extension target.

This is definitely not ideal and will double the compilation time since the dart code is compiling twice, once for the Pod and again for the app.

We should make this much, much easier...

**Comment by jmagman** [comment]
This is similar how add-to-app works where the compiled dart code App.framework is actually embedded into the host app via CocoaPods:
https://github.com/flutter/flutter/blob/4f964196128e60adff006409868348d4527b3b9f/packages/flutter_tools/templates/module/ios/library/Flutter.tmpl/podhelper.rb.tmpl#L93-L95

**Comment by rafaelsetragni** [comment]
I got it. But I found an error in this mean time. If we use the flutter debug and release configurations, the target extension is not being able to start by it own without a debugger attached, similar of what happen to the main app without debug connection. Because of it, the target extension are not being capable to wake up on debug mode.

To solve it I preserved the original build configurations and removed the #FLUTTER_ROOT variable from script phase and I concatenated the result string inside pod file, with the local variable flutter_root.

![image](https://user-images.githubusercontent.com/40064496/168407339-26b4b0fb-b98b-42ba-b473-a94f7d77ec9c.png)

```ruby
def update_awesome_fcm_service_target(xcodeproj_path, flutter_root)
     target_name = 'AwesomeServiceExtension'
     build_phase_name = "Awesome Notifications Build Phase"
     flutter_build_script = '/bin/sh "'+flutter_root+'/packages/flutter_tools/bin/xcode_backend.sh" build'
     desired_phase_index = 1
```

> We should make this much, much easier...

Exactly! As simple as possible. But so far, I've only could achieve it modifying the PodFile.

**Comment by jmagman** [comment]
Did the ` s.script_phase` podspec code I suggested not work for you?  It worked for me with just a fresh Flutter app, a new notification target, and that podspec change.

**Comment by jmagman** [comment]
>  If we use flutter's debug and release build configurations

Meaning:  don't change anything in the user's project or Podfile, just change the `awesome_notifications` podspec with the line I suggested, and that's it.

**Comment by rafaelsetragni** [comment]
I gonna try it right now! Hold on.

**Comment by rafaelsetragni** [comment]
No, it didn't add the script build phase, even cleaning the project or/and removing all the pod modifications.

But I think a single modification at the end of the Podfile is simple enough to release to the public. It's just copy and paste, hopefully developers won't run into problems at this step. To minimize future problems, I'm going to provide then a good step by step, showing how to do these extra steps on iOS and how to make sure that everything is in the right place.

**Comment by jmagman** [comment]
> No, it didn't add the script build phase, even cleaning the project or/and removing all the pod modifications.

Are you sure?  It should have added it to the awesome_notification target:
<img width="1232" alt="Screen Shot 2022-05-16 at 4 39 27 PM" src="https://user-images.githubusercontent.com/682784/168699281-4a75af0b-9e46-4b36-92f4-53521dadc4be.png">

**Comment by rafaelsetragni** [comment]
Yes! You are right! It was added into the Pods, not the main project.

**Comment by rafaelsetragni** [comment]
![image](https://user-images.githubusercontent.com/40064496/168699693-f54047f5-7c42-477b-8eae-47375fb2b224.png)

But without the `target 'AwesomeServiceExtension' do`, the plugin dependencies are not being added. Is there a way to add this dependencies on pod spec as well?

**Comment by jmagman** [comment]
> But without the target 'AwesomeServiceExtension' do, the plugin dependencies are not being added. Is there a way to add this dependencies on pod spec as well?

No you still need that, as I said: 
> Then your users don't need to do anything in their Podfile except pod awesome_notifications in the extension target.

**Comment by rafaelsetragni** [comment]
Yes, but I think it's good enough. And even simpler than before!
Thank you very much @jmagman for your help!

**Comment by jmagman** [comment]
We need to keep this open, this isn't a good solution, and at a minimum we should document it.

**Comment by wangweiarr** [comment]
I have the same problem, keeping watch this.

**Comment by wangweiarr** [comment]
I have the same problem, keeping watch this.

**Comment by mlb5000** [comment]
I am getting this as well. But ONLY on MacOS, and only for the Release build configuration...

**Comment by KoheiKanagu** [comment]
Setting OTHER_LDFLAGS in Extension to an empty string may solve the problem.
<img width="682" alt="スクリーンショット 2024-01-05 19 07 45" src="https://github.com/flutter/flutter/assets/6175794/f57b1397-cf0a-462c-b1f8-491437e56c64">
<img width="681" alt="スクリーンショット 2024-01-05 19 07 49" src="https://github.com/flutter/flutter/assets/6175794/c8b93de0-923f-4a18-8a28-5e8081daef39">

**Comment by jmagman** [comment]
@KoheiKanagu I'm going to hide your message, that's not a good solution. 🙂 

@mlb5000 check out the second suggestion in https://github.com/flutter/flutter/issues/103344#issuecomment-1125497816 but you'll have to change it for macOS, you can ignore the simulator stuff.

> Profile and Release > Any iOS SDK add "${FLUTTER_ROOT}/bin/cache/artifacts/engine/ios-release/Flutter.xcframework/ios-arm64"

For macOS this would be adding to `FRAMEWORK_SEARCH_PATHS` 
Debug: `${FLUTTER_ROOT}/bin/cache/artifacts/engine/darwin-x64/`
Profile: `${FLUTTER_ROOT}/bin/cache/artifacts/engine/darwin-x64-profile/`
Release: `${FLUTTER_ROOT}/bin/cache/artifacts/engine/darwin-x64-release/`

![Screenshot 2024-01-05 at 11 41 00 AM](https://github.com/flutter/flutter/assets/682784/a140483b-aa42-4230-bdeb-56d7ec5afdb8)

Does that work?  Sorry this is an undocumented mess...
